lib/api.ts

```diff
@@ -98,7 +98,28 @@ export async function getContributorBySlug(file: string, detail = false) {
   }
 
   const slackMessages = await getSlackMessages(data.slack);
-
+  const UtctoDate = (date: number | string) => {
+    let dateObject;
+    if (
+      data.joining_date !== "Still waiting for this" &&
```

Comment:
In what situations would `joining_date` hold a value of `"Still waiting for this"`?

```suggestion
      data.joining_date !== "Still waiting for this" &&
```

Reply:
[
<img width="1168" alt="Screenshot 2024-02-29 at 1 21 14 PM" src="https://github.com/coronasafe/leaderboard/assets/115660976/9eeb235e-4a57-4bbd-8170-436996db15fb">
](url)

 when there is no  .json file made of the user we go to the .md file. When joining_date is not there  in the .md file  we use the given condition.

Reply:
@rithviknishad

Reply:
@rithviknishad  we are not relying  joining_date we only rely on .json of a user only when the .json file is not created we check the .md file and try to get the date.

Reply:
<img width="854" alt="image" src="https://github.com/coronasafe/leaderboard/assets/25143503/3f45df02-6a70-4baa-8ea2-c64a5bda5f9d">

If there's no activity for a user, let's keep the `first_activity` as null. 
`joining_date` is NOT the date of first activity of a contributor.

Reply:
okay  got it  thanks.

Reply:
made the required changes please check  @rithviknishad ..